pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -178,7 +179,12 @@ func (configgen *ConfigGeneratorImpl) buildClusters(proxy *model.Proxy, req *mod
 		}
 		clusters = append(clusters, patcher.insertedClusters()...)
 	}
-
+	// if SDS_ADDR is defined and is using UDS, create a cluster for it
+	// if SDS_ADDR is defined and is using service name, there should already a cluster added via service discovery
+	externalSDSAddr := features.SDSEndpoint
+	if externalSDSAddr != "" && strings.HasPrefix(externalSDSAddr, authn_model.UnixAddressPrefix) {
```

Comment:
Confused - what is the unix address doing ? 

There are 2 choices, either istiod configs using a cluster for the real SDS server, from envoy - which is a bit more complicated but probably a good idea, in which case there is no unix address involved.

Or Istiod doesn't even know about the SDS_ADDR - except injection template. And pilot-agent uses the SDS_ADDR internally to create a tunnel, like it does for Istiod. 

It seems you went with the first option, so no unix address ?

Reply:
Or 3rd choice - don't touch the agent, don't create a new UDS socket - instead agent talks with Istiod like before, and Istiod is federating.